threadpoolctl.py

```diff
@@ -927,6 +971,7 @@ def __len__(self):
 
     def _load_libraries(self):
         """Loop through loaded shared libraries and store the supported ones"""
+        self.lib_controllers = []  # reset the list of controllers
```

Comment:
This will be a bit weird in the case of flexiblas: when we switch backend, the controller instance of flexiblas itself will no longer be the controller used by the caller to trigger the switch.

Maybe this is not that a big problem but it might be a bit surprising and maybe cause weird bugs.

The alternative would be to not reset the previously loaded controllers and change `_make_controller_from_path` to only append to the list of controllers controllers that match previously unseen filepaths.

Reply:
Good point !

> The alternative would be to not reset the previously loaded controllers and change _make_controller_from_path to only append to the list of controllers controllers that match previously unseen filepaths.

I did that